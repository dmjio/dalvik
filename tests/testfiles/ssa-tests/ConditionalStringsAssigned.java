class ConditionalStringsAssigned {
  static Object target;
  static Object entry(int x) {
    String s;
    if(x > 0)
      s = "foo";
    else
      s = "bar";

    target = s;
    return s;
  }
}
