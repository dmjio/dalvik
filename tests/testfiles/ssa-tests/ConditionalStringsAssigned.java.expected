SimplePhi { numBlocks = 4
          , phis = [(0, fromList ["foo", "bar"])]
          , blockPhis = [(0, 2)]
          , assignedPhis = [(0, "target")]
          }